pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -439,3 +439,23 @@ func (o *LoginOptions) serverProvided() bool {
 func (o *LoginOptions) tokenProvided() bool {
 	return len(o.Token) > 0
 }
+
+func (o *LoginOptions) ConfigIsWritable() (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	info, err := os.Stat(kubeconfig)
+	if err != nil {
+		if os.IsNotExist(err) {
+			info, err = os.Stat(filepath.Dir(kubeconfig))
+			if err != nil {
+				return false, err
+			}
+		} else if os.IsPermission(err) {
+			return false, nil
+		} else {
+			return false, err
+		}
+	}
+
+	return info.Mode().Perm()&0200 != 0, nil
```

Comment:
Might be nice to place the bitwise operation in a function or at least an explanatory variable. Likewise to define `0200` as a constant to name it.